editor/components/provider/index.js

```diff
@@ -22,73 +22,37 @@ import {
 import { setupEditor, undo, redo, createUndoLevel } from '../../store/actions';
 import store from '../../store';
 
-/**
- * The default editor settings
- * You can override any default settings when calling initializeEditor
- *
- *  alignWide   boolean   Enable/Disable Wide/Full Alignments
- *
- * @var {Object} DEFAULT_SETTINGS
- */
-const DEFAULT_SETTINGS = {
-	alignWide: false,
-	colors: [
-		'#f78da7',
-		'#cf2e2e',
-		'#ff6900',
-		'#fcb900',
-		'#7bdcb5',
-		'#00d084',
-		'#8ed1fc',
-		'#0693e3',
-		'#eee',
-		'#abb8c3',
-		'#313131',
-	],
-
-	// This is current max width of the block inner area
-	// It's used to constraint image resizing and this value could be overriden later by themes
-	maxWidth: 608,
-
-	// Allowed block types for the editor, defaulting to true (all supported).
-	blockTypes: true,
-};
-
 class EditorProvider extends Component {
 	constructor( props ) {
 		super( ...arguments );
 
 		this.store = store;
 
-		this.settings = {
-			...DEFAULT_SETTINGS,
-			...props.settings,
-		};
-
 		// Assume that we don't need to initialize in the case of an error recovery.
 		if ( ! props.recovery ) {
-			this.store.dispatch( setupEditor( props.post, this.settings ) );
-		}
-	}
-
-	getChildContext() {
-		return {
-			editor: this.settings,
-		};
-	}
-
-	componentWillReceiveProps( nextProps ) {
-		if (
-			nextProps.settings !== this.props.settings
-		) {
-			// eslint-disable-next-line no-console
-			console.error( 'The Editor Provider Props are immutable.' );
+			this.store.dispatch(
+				setupEditor( props.post, {
+					...EditorSettings.defaultSettings,
```

Comment:
Something about this feels off to me. Partly because we're duplicating the initialization of these settings between here and below. I think also it's odd that we don't need to observe changes in settings here; why does this only need to occur at setup time, and never thereafter? Should this be a proper `EditorSettings.Consumer` so we can just leverage the value from `EditorSettings.Provider` directly?

Reply:
Noted in the description :) (Second point)

Reply:
Ah, I did read it, apparently not closely enough.